inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -61,7 +61,64 @@ IE::IExecutableNetworkInternal::Ptr AutoInferencePlugin::LoadExeNetworkImpl(cons
     }
 
     auto networkPrecision = GetNetworkPrecision(network);
-    return LoadNetworkImpl(network, config, networkPrecision);
+    return LoadNetworkImpl({}, network, config, networkPrecision);
+}
+
+std::shared_ptr<AutoExecutableNetwork> AutoInferencePlugin::LoadNetworkImpl(const std::string& modelPath,
+                                                                            const InferenceEngine::CNNNetwork& network,
+                                                                            const ConfigType& config,
+                                                                            const std::string& networkPrecision) {
+    if (GetCore() == nullptr) {
+        IE_THROW() << "Please, work with AUTO device via InferencEngine::Core object";
+    }
+
+    if (modelPath.empty() && network.getFunction() == nullptr) {
+        IE_THROW() << "AUTO device supports just ngraph network representation";
+    }
+
+    auto fullConfig = mergeConfigs(_config, config);
+    CheckConfig(fullConfig);
+    auto metaDevices = GetDeviceList(fullConfig);
+    auto core = GetCore();
+    auto LoadNetworkAsync =
+        [core, modelPath, network](const std::string& device)
+            -> IE::SoExecutableNetworkInternal {
+            IE::SoExecutableNetworkInternal executableNetwork;
+            std::cout << "!!! DEBUG: Starting Async loading to the " << device <<  " !!!" << std::endl;
+            std::cout << "!!! DEBUG: device full name: " << core->GetMetric(device, METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>() << std::endl;
+            if (!modelPath.empty()) {
+                executableNetwork = core->LoadNetwork(modelPath, device, {});
+            } else {
+                executableNetwork = core->LoadNetwork(network, device, {});
+            }
+            std::cout << "!!! DEBUG: " << device << " was loaded !!!" << std::endl;
+            return executableNetwork;
+        };
+
+    NetworkFuture cpuFuture;
+    NetworkFuture acceleratorFuture;
+
+    // start CPU task
+    const auto CPUIter = std::find_if(metaDevices.begin(), metaDevices.end(),
+                                      [=](const std::string& d)->bool{return d.find("CPU") != std::string::npos;});
+    if (CPUIter != metaDevices.end()) {
+        // not use std::launch::async, so let OS decide delay or async.
+        cpuFuture = std::async(std::launch::async, LoadNetworkAsync, *CPUIter);
+    }
+
+    // start accelerator task, like GPU
+    const auto accelerator = SelectDevice(metaDevices, networkPrecision);
+    bool isAccelerator = accelerator.find("CPU") == std::string::npos;
+    if (isAccelerator) {
+        // not use std::launch::async, so let OS decide delay or async.
```

Comment:
the comment is incorrect :)

Reply:
yes, will modify. BTW do you think should we use the default "std::launch::async | std::launch::deferred" ?